clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/deploy/ops/GoogleUserDataProvider.groovy

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright 2017 Google, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.clouddriver.google.deploy.ops
+
+import com.netflix.frigga.Names
+import com.netflix.spinnaker.clouddriver.google.deploy.description.BasicGoogleDeployDescription
+import com.netflix.spinnaker.clouddriver.google.security.GoogleNamedAccountCredentials
+import groovy.transform.PackageScope
+import groovy.util.logging.Slf4j
+import org.springframework.stereotype.Component
+
+
+/**
+ * Provides the common user data from a local file to be applied to all Google deployments.
+ *
+ * Any custom user data specified for each deployment will be appended to common user data, allowing custom user data
+ * to override the common user data.
+ */
+@Slf4j
+@Component
+public class GoogleUserDataProvider {
+
+  GoogleUserDataProvider() {
+  }
+
+  /**
+   * Returns the user data as a Map.
+   */
+  Map getUserData(final String serverGroupName, final String instanceTemplateName,
+                  BasicGoogleDeployDescription description,
+                  GoogleNamedAccountCredentials credentials, String customUserData) {
+    String userDataFile = credentials.getUserDataFile()
+    String rawUserData = getFileContents(userDataFile)
+    String commonUserData = replaceTokens(rawUserData, description, serverGroupName, instanceTemplateName)
+
+    def userDataMap = stringToUserDataMap(commonUserData)
+
+    if (customUserData){
+      def customUserDataMap = stringToUserDataMap(customUserData)
```

Comment:
Any value in performing `replaceTokens` on the `customUserData` as well? It might be nice for operations like clone to have userdata like `%%autogrp%%` always up-to-date. @duftler